huaweicloud/services/cbr/resource_huaweicloud_cbr_replicate_backup.go

```diff
@@ -0,0 +1,168 @@
+package cbr
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var replicateBackupNonUpdatableParams = []string{
+	"backup_id",
+	"replicate.0.destination_project_id",
+	"replicate.0.destination_region",
+	"replicate.0.destination_vault_id",
+	"replicate.0.name",
+	"replicate.0.description",
+	"replicate.0.enable_acceleration",
+}
+
+// @API CBR POST /v3/{project_id}/backups/{backup_id}/replicate
+func ResourceReplicateBackup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceReplicateBackupCreate,
+		ReadContext:   resourceReplicateBackupRead,
+		UpdateContext: resourceReplicateBackupUpdate,
+		DeleteContext: resourceReplicateBackupDelete,
+
+		CustomizeDiff: config.FlexibleForceNew(replicateBackupNonUpdatableParams),
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"backup_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"replicate": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"destination_project_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"destination_region": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"destination_vault_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"enable_acceleration": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"enable_force_new": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"true", "false"}, false),
+				Description:  utils.SchemaDesc("", utils.SchemaDescInput{Internal: true}),
+			},
+		},
+	}
+}
+
+func buildReplicateBackupBodyParams(d *schema.ResourceData) map[string]interface{} {
+	replicateRaw := d.Get("replicate").([]interface{})
+	if len(replicateRaw) == 0 {
+		return nil
+	}
+
+	bodyParams := map[string]interface{}{
+		"replicate": map[string]interface{}{},
+	}
+
+	replicate := replicateRaw[0].(map[string]interface{})
```

Comment:
Direct type conversion is risky. If the script contains replicate {null}, this line of code will panic.